src/Controller/ModulesController.php

```diff
@@ -334,4 +365,104 @@ public function upload()
             'id' => $response['data']['id'],
         ]);
     }
+
+    /**
+     * Export data in csv format
+     *
+     * @return \Cake\Http\Response
+     */
+    public function changeStatus() : Response
+    {
+        $this->request->allowMethod(['post']);
+        $ids = $this->request->getData('ids');
+        $status = $this->request->getData('status');
+        if (!empty($ids)) { // export selected (filter by id)
+            $ids = explode(',', $ids);
+            foreach ($ids as $id) {
+                $data = [
+                    'id' => $id,
+                    'status' => $status,
+                ];
+                $this->apiClient->saveObject($this->objectType, $data);
+            }
+        }
+
+        return $this->redirect(['_name' => 'modules:list', 'object_type' => $this->objectType]);
+    }
+
+    /**
+     * Export data in csv format
+     *
+     * @return void
+     */
+    public function export() : void
+    {
+        $this->request->allowMethod(['post']);
+        $fields = ['id', 'title', 'description', 'uname', 'status', 'body', 'lang', 'extra'];
+        $delimiter = ',';
+        $ids = $this->request->getData('ids');
+        $csv = implode($delimiter, $fields) . "\n";
+        if (!empty($ids)) { // export selected (filter by id)
+            $response = $this->apiClient->getObjects($this->objectType, ['filter' => ['id' => $ids]]);
+            $csv .= $this->csvRows($response, $fields, $delimiter);
+            $csv = rtrim($csv, "\n"); // remove last newline
+            $this->set('csv', $csv);
+        } else { // export all
+            $response = $this->apiClient->getObjects($this->objectType, [
+                'page' => 1,
+                'page_size' => 100,
+            ]);
+            while (!empty($response['links']['next'])) {
```

Comment:
I'm not sure of the consequences of an `unlimited` export loop.
We may use just `'ids'` for now and think about a limit later.

Reply:
For now, only "checked" items export is allowed, via post. I'll drop "unlimited" items export part...